### PR TITLE
Bugs

### PR DESCRIPTION
**Pull request by jicata**

### CODE REVIEW

File: Open Judge System/Web/OJS.Web/Controllers/SubmissionsController.cs [hunk 1]
```diff
@@ -66,7 +67,20 @@ public ActionResult Index()
             bool notProcessedOnly = false,
             int? contestId = null)
         {
-            var data = this.User.IsAdmin() ? this.Data.Submissions.All() : this.Data.Submissions.AllPublic();
+            IQueryable<Submission> data = null;
+
+            if (this.User.IsLecturer())
+            {
+                data = this.Data.Submissions.AllForLecturer(this.UserProfile.Id);
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Controllers/SubmissionsController.cs]: Can we change the name of the method to something more descriptive like AllPublicWithLecturerContests?

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs [hunk 2]
```diff
@@ -598,5 +630,18 @@ private bool IsBestSubmission(int problemId, int participantId, int submissionId
...
         }
+
+        private bool IsSubmissionProcessing(Submission submission)
+        {
+            if (submission.Processed)
+            {
+                return false;
+            }
+
+            var submissionForProcessing = this.Data.SubmissionsForProcessing.All()
```
Comment (rozay) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs]: Is this some new convention `.All()` to be on the upper line?

**Comment by rozay** [comment]
@jicata Please refer and describe the issues this PR solves. Thanks :)

File: Open Judge System/Data/OJS.Data/Repositories/SubmissionsRepository.cs [hunk 3]
```diff
@@ -14,38 +16,45 @@ public SubmissionsRepository(IOjsDbContext context)
...
-                    (((x.Participant.IsOfficial && x.Problem.Contest.ContestPassword == null) ||
-                     (!x.Participant.IsOfficial && x.Problem.Contest.PracticePassword == null))
-                    && x.Problem.Contest.IsVisible && !x.Problem.Contest.IsDeleted
-                    && x.Problem.ShowResults));
-        }
+        public IQueryable<Submission> AllPublic() => 
+            this.All()
+                .Where(s => s.IsPublic ??
+                    (((s.Participant.IsOfficial && s.Problem.Contest.ContestPassword == null) ||
+                    (!s.Participant.IsOfficial && s.Problem.Contest.PracticePassword == null)) &&
```
Comment (rozay) [inline Open Judge System/Data/OJS.Data/Repositories/SubmissionsRepository.cs]: IMO `tab`

File: Open Judge System/Data/OJS.Data/Repositories/SubmissionsRepository.cs [hunk 4]
```diff
@@ -14,38 +16,45 @@ public SubmissionsRepository(IOjsDbContext context)
...
+                    s.Problem.Contest.IsVisible && !s.Problem.Contest.IsDeleted &&
```
Comment (rozay) [inline Open Judge System/Data/OJS.Data/Repositories/SubmissionsRepository.cs]: Every condition on new line

File: Open Judge System/Data/OJS.Data/Repositories/SubmissionsRepository.cs [hunk 5]
```diff
@@ -14,38 +16,45 @@ public SubmissionsRepository(IOjsDbContext context)
...
+        public IQueryable<Submission> AllPublicWithLecturerContests(string lecturerId)
         {
-            var submission =
+            var problemsIds = new HashSet<int>(
+                this.Context.Contests
+                    .Where(c => c.Category.Lecturers.Any(cat => cat.LecturerId == lecturerId) ||
+                        c.Lecturers.Any(l => l.LecturerId == lecturerId))
+                    .SelectMany(c => c.Problems.Select(p => p.Id)));
+
+            var submissions = this.All()
```
Comment (rozay) [inline Open Judge System/Data/OJS.Data/Repositories/SubmissionsRepository.cs]: `.All()` on new line

File: Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs [hunk 6]
```diff
@@ -202,18 +199,47 @@ public ActionResult Update(ModelType model)
...
+                            this.Data.SaveChanges();
+
+                            var submissionIsBestSubmission = this.IsBestSubmission(
+                                submissionProblemId,
+                                submissionParticipantId,
+                                submission.Id);
+
+                            if (submissionIsBestSubmission)
+                            {
+                                this.Data.ParticipantScores.RecalculateParticipantScore(submissionParticipantId, submissionProblemId);
```
Comment (rozay) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/SubmissionsController.cs]: This line is more than 120 characters. `.RecalculateParticipantScore(submissionParticipantId, submissionProblemId);` on enw line

**Pull request merged**